backend/segmentation/api/tracker_kymograph.py

```
@@ -429,30 +432,75 @@ def _build_link_cost(
     if P == 0 or Q == 0:
         return np.zeros((P, Q), dtype=np.float64)
 
-    demb = np.full((P, Q), np.nan, dtype=np.float64)
-    for i in range(P):
-        for j in range(Q):
-            d = _emb_distance(prev_feats[i], nxt_feats[j])
-            if d is not None:
-                demb[i, j] = d
+    # Build the whole P×Q matrix with broadcasted numpy ops rather than a
```

Ruff (0.15.20)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)